src/gui/folder.cpp

```diff
@@ -166,18 +165,42 @@ void Folder::checkLocalPath()
 
     if (fi.isDir() && fi.isReadable()) {
         qCDebug(lcFolder) << "Checked local path ok";
+        return true;
     } else {
         // Check directory again
         if (!FileSystem::fileExists(_definition.localPath, fi)) {
-            _syncResult.appendErrorString(tr("Local folder %1 does not exist.").arg(_definition.localPath));
-            _syncResult.setStatus(SyncResult::SetupError);
+            if (ocApp()) {
+                if (FileSystem::fileExists(fi.path() + "/..")) {
+                    int button = QMessageBox::question(ocApp()->gui()->settingsDialog(),
+                        tr("Your %1 folder %2 does not exist.").arg(ocApp()->applicationName(), _definition.localPath),
+                        tr("The folder %1 does not exist, do you want to create it or quit %2.").arg(_definition.localPath, ocApp()->applicationName()),
+                        tr("Create Folder"), tr("Quit"));
+                    if (button == 0) {
+                        if (!QDir(_definition.localPath + "/..").mkdir(fi.dir().dirName())) {
+                            qCWarning(lcFolder) << "Failed to creat sycn root" << _canonicalLocalPath << fi.dir().dirName();
+                        }
+                        // give the system some time, vfs might crash
+                        QThread::sleep(1);
+                        return checkLocalPath();
+                    }
+                } else {
+                    QMessageBox::warning(ocApp()->gui()->settingsDialog(),
+                        tr("Your %1 folder %2 does not exist.").arg(ocApp()->applicationName(), _definition.localPath),
+                        tr("The folder %1 and its parent folder do not exist. %2 will quit now.").arg(_definition.localPath, ocApp()->applicationName()));
+                }
+                _syncResult.setStatus(SyncResult::SetupError);
+                qApp->quit();
```

Comment:
Do we need to quit here?

Can't we disable the account or something?

This way you can't remove an offending account if you can't recreate the folder on disk?

Reply:
We ignored the folder before and it went magically back online once a folder appeared. https://github.com/owncloud/enterprise/issues/4048
And in general I guess most people will only have one account.
That you can't remove the account yes that needs to be handled....

Reply:
I am not quite getting why you remove the account at all, instead of just removing the sync folder. But I am sure you thought about that... ;-)